### PR TITLE
New package: pam_zfscrypt 0.2.0

### PR DESCRIPTION
**Pull request by zdykstra**

### CODE REVIEW

**Comment by zdykstra** [comment]
It's currently beyond me to fix the the cross-compiling issues present here.

**Comment by ndowens** [comment]
I will check and see if I can get it to cross

**Comment by q66** [comment]
as far as i can see you shouldn't need `zfs-devel` in `hostmakedepends`

if anything from zfs is needed in the host system during build, you should probably use `zfs` instead, but chances are it's not

**Comment by ndowens** [comment]
> as far as i can see you shouldn't need `zfs-devel` in `hostmakedepends`
> 
> if anything from zfs is needed in the host system during build, you should probably use `zfs` instead, but chances are it's not

It complains about missing header if cross-compiling

**Comment by zdykstra** [comment]
> [ndowens@8d4f530](https://github.com/ndowens/void-packages/commit/8d4f530f62875478d2abbce5c25f3b88629af79d)
> Result:
> https://travis-ci.org/ndowens/void-packages/builds/621908137
> 
> This is how I did it and seems to work

Thanks for the help!  This should be good to go now.

File: srcpkgs/pam_zfscrypt/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'pam_zfscrypt'
+pkgname=pam_zfscrypt
+version=0.2.0
+revision=1
+wrksrc="zfscrypt-${version}"
+build_style=gnu-makefile
+make_use_env=yes
+hostmakedepends="zfs-devel"
```
Comment (abenson) [inline srcpkgs/pam_zfscrypt/template]: move to makedepends=

File: srcpkgs/pam_zfscrypt/INSTALL.msg [hunk 2]
```diff
@@ -0,0 +1 @@
+Refer to https://github.com/BenKerry/zfscrypt#setup for configuration details.
```
Comment (Duncaen) [inline srcpkgs/pam_zfscrypt/INSTALL.msg]: Don't use `INSTALL.msg` for things like that, they are already not being read and adding more doesn't improve the situation.

File: srcpkgs/pam_zfscrypt/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+makedepends="pam-devel zfs-devel"
+make_dirs="/usr/lib/security 0755 root root"
+depends="zfs pam"
+short_desc="Automatically decrypt and mount encrypted ZFS home directories"
+maintainer="Zach Dykstra <dykstra.zachary@gmail.com>"
+license="MIT"
+homepage="https://github.com/BenKerry/zfscrypt"
+distfiles="https://github.com/BenKerry/zfscrypt/archive/v${version}.tar.gz"
+checksum=@032766fca0f4d6da47faa29a92195f8357f2e8073e49e14c9cb719b3248c3a88
```
Comment (Duncaen) [inline srcpkgs/pam_zfscrypt/template]: Why does this use the checksum of files inside the archive and not the archive itself?

File: srcpkgs/pam_zfscrypt/template [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'pam_zfscrypt'
+pkgname=pam_zfscrypt
+version=0.2.0
+revision=1
+wrksrc="zfscrypt-${version}"
+build_style=gnu-makefile
+make_use_env=yes
+hostmakedepends="zfs-devel"
+makedepends="pam-devel zfs-devel"
+make_dirs="/usr/lib/security 0755 root root"
```
Comment (Duncaen) [inline srcpkgs/pam_zfscrypt/template]: This looks wrong, the package contains files in that directory there is no need to create it.

File: srcpkgs/pam_zfscrypt/template [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+depends="zfs pam"
```
Comment (abenson) [inline srcpkgs/pam_zfscrypt/template]: Should let the auto dependency magic happen

**Comment by xtraeme** [comment]
what others said, please check their comments

File: srcpkgs/pam_zfscrypt/template [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+short_desc="Automatically decrypt and mount encrypted ZFS home directories"
+maintainer="Zach Dykstra <dykstra.zachary@gmail.com>"
+license="MIT"
+homepage="https://github.com/BenKerry/zfscrypt"
+distfiles="https://github.com/BenKerry/zfscrypt/archive/v${version}.tar.gz"
+checksum=@032766fca0f4d6da47faa29a92195f8357f2e8073e49e14c9cb719b3248c3a88
+noverifyrdeps=yes
```
Comment (abenson) [inline srcpkgs/pam_zfscrypt/template]: Don't do this, add zfs's libs to common/shlibs 

    /usr/lib/libzfs.so.2 zfs-0.8.2_1
    /usr/lib/libnvpair.so.1 zfs-0.8.2_1

etc

Comment (abenson) [inline srcpkgs/pam_zfscrypt/template]: Don't do this, add zfs's libs to common/shlibs 

    /usr/lib/libzfs.so.2 zfs-0.8.2_1
    /usr/lib/libnvpair.so.1 zfs-0.8.2_1

etc

File: srcpkgs/pam_zfscrypt/template (hunk 3 above)
Comment (zdykstra) [reply srcpkgs/pam_zfscrypt/template]: Incorrect use of xgensum, fixed!

File: srcpkgs/pam_zfscrypt/template (hunk 6 above)
Comment (zdykstra) [reply srcpkgs/pam_zfscrypt/template]: Added to common/shlibs

File: srcpkgs/pam_zfscrypt/INSTALL.msg (hunk 2 above)
Comment (zdykstra) [reply srcpkgs/pam_zfscrypt/INSTALL.msg]: Noted. Removed!

**Comment by zdykstra** [comment]
https://travis-ci.org/void-linux/void-packages/builds/624705727

That is without `zfs-devel` in `hostmakedepends`.

**Comment by ndowens** [comment]
I think I will have another way you can do this without having zfs-devel in hostmakedepends

**Comment by zdykstra** [comment]
Barring a change to the package name and hostmakedepends being there, I believe all of the issues so far have been addressed. Please let me know if any further changes are required.

**Comment by ndowens** [comment]
Ok believe I got it where there is no need of having zfs-devel in host

**Comment by zdykstra** [comment]
> Ok believe I got it where there is no need of having zfs-devel in host

Nice - I can bump the package if you can let me know what to change.

**Comment by ndowens** [comment]
> > Ok believe I got it where there is no need of having zfs-devel in host
> 
> Nice - I can bump the package if you can let me know what to change.

No need to bump the revision, since it is not in repo. Ok here is the diff I made:
https://termbin.com/1s036

**Comment by ndowens** [comment]
> > Ok believe I got it where there is no need of having zfs-devel in host
> 
> Nice - I can bump the package if you can let me know what to change.

No need to revbump, since it is not in repo 

Here it is : https://termbin.com/ypt1

**Comment by ndowens** [comment]
There we go, built perfectly it seems

**Comment by zdykstra** [comment]
> There we go, built perfectly it seems

I really appreciate your help on this package. Thank you for taking the time to fix the build issues.

**Comment by ndowens** [comment]
> > There we go, built perfectly it seems
> 
> I really appreciate your help on this package. Thank you for taking the time to fix the build issues.

Just noticed -fPIC is still in CFLAGS, since we used make_use_env=yes should not be needed anymore(fPIC)

File: srcpkgs/pam_zfscrypt/template [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'pam_zfscrypt'
+pkgname=pam_zfscrypt
+version=0.2.0
+revision=1
+wrksrc="zfscrypt-${version}"
+build_style=gnu-makefile
+make_use_env=yes
+makedepends="pam-devel zfs-devel"
+depends="zfs pam"
```
Comment (xtraeme) [inline srcpkgs/pam_zfscrypt/template]: `zfs` dependency is pulled in via xbps-src, please remove this.

File: srcpkgs/pam_zfscrypt/template (hunk 6 above)
[the remark this answers is not in the document]
Comment (zdykstra) [reply srcpkgs/pam_zfscrypt/template]: Already been removed, fixed in common/shlibs

File: srcpkgs/pam_zfscrypt/template (hunk 7 above)
Comment (zdykstra) [reply srcpkgs/pam_zfscrypt/template]: Removed!

**Pull request merged**